src/Framework/Profiler/EvaluationLocation.cs

```diff
@@ -93,94 +99,135 @@ public struct EvaluationLocation
         public string ElementName { get; }
 
         /// <nodoc/>
-        public string Description { get; }
+        public string ElementDescription { get; }
 
         /// <nodoc/>
         public EvaluationLocationKind Kind { get; }
 
         /// <nodoc/>
-        public static EvaluationLocation CreateLocationForCondition(EvaluationPass evaluationPass, string evaluationDescription, string file,
+        public bool IsEvaluationPass => File == null;
```

Comment:
Hopefully this will always hold :)

Reply:
That's the contract so far

---
In reply to: [157258488](https://github.com/Microsoft/msbuild/pull/2813#discussion_r157258488) [](ancestors = 157258488)